Lombiq.Hosting.Tenants.FeaturesGuard/Handlers/FeaturesEventHandler.cs

```diff
@@ -92,11 +92,14 @@ public async Task EnableConditionallyEnabledFeaturesAsync(IFeatureInfo featureIn
 
         // If Shell Descriptor's Features already contains a feature that is found in conditionalFeatures, remove it
         // from the list.
+        // Double enumeration is necessary, since we can't remove from the List without breaking enumeration.
+#pragma warning disable CA1851 // Possible multiple enumerations of 'IEnumerable' collection
         var featuresToEnable = conditionalFeatures.ToList();
         foreach (var feature in conditionalFeatures.Where(feature => shellDescriptor.Features.Contains(new ShellFeature(feature.Id))))
         {
             featuresToEnable.Remove(feature);
         }
+#pragma warning restore CA1851 // Possible multiple enumerations of 'IEnumerable' collection
```

Comment:
You actually have multiple enumeration and I don't think it's really necessary, so the warning shouldn't be disabled.

It's not clear to me why `var conditionalFeatures = ...` has to be before `var shellDescriptor = ...`, is that really necessary? I don't see why. If not, then you could switch their places, and then `conditionalFeatures` and `featuresToEnable` can be merged into a single expression, as in `allFeatures.Where(feature => ... && !...).ToList()`.

Reply:
Indeed!